lib/ffmpeg/vaapi/decoder.py

```diff
@@ -77,5 +77,14 @@ def check_output(self):
       "hwaccel initialisation returned error", self.output, re.MULTILINE)
     assert m is None, "Failed to use hardware decode"
 
+    # AV1d specific (checked here to ensure we cover all bit-depth tests for AV1d,
+    # current and future).
+    #
+    # Don't allow false success if ffmpeg falls back to a SW alternative... it
+    # clearly defeats the purpose of our test.
+    m = re.search(
+      "Your platform doesn't suppport hardware accelerated AV1 decoding", self.output, re.MULTILINE)
```

Comment:
Yes, agree.  I will update it to increased _potential_ future scope.

But this still doesn't guarantee we catch future test escapes.  Because anyone can change the debug message to something completely different, at any time.  We really need a **"standard"** message to detect HW failure ---> SW fallback.  Or an option to force ffmpeg to avoid any fallbacks and abort instead.

@xhaihao do you know a way to force ffmpeg to avoid fallback's if HW fails?  Maybe we need to develop a patch?  Or we at least need to "standardize" the log message.  We keep chasing these message variations over and over again.

Reply:
Nice catch, btw 😃

Reply:
AFAIK there is no option to disable the fallback.

Reply:
Ok.  Other ideas to improve the detection are welcome!

Reply:
> Can we control the log level of those message?

IDK, it's inconsistent all over the place.  It needs better standardization.

Reply:
UGH! Forgot to expand message scope before merging.  Without standardization or other stop-gaps, it's still better then nothing and catches "today's" warnings.  I will update in another patch.